tests/app/main/test_services.py

```diff
@@ -50,129 +51,289 @@ def supplier_remove_service__post_data(request):
+                supplier_id=1234,
+                framework="g-cloud-909",
+            )
+
+    @pytest.mark.parametrize("framework_slug", (
+        "g-cloud-890",
+        "g-cloud-870",
+        "x-cloud-123",
+    ))
+    @mock.patch('app.main.views.services.data_api_client')
+    def test_no_services_list_inappropriate_frameworks(self, data_api_client, framework_slug):
+        with self.app.test_client():
+            self.login()
+
+            data_api_client.get_framework.side_effect = self._mock_get_framework_side_effect
+            data_api_client.find_services.side_effect = AssertionError("Shouldn't be called")
+
+            res = self.client.get("/suppliers/frameworks/{}/services".format(framework_slug))
+            assert res.status_code == 404
 
     @mock.patch('app.main.views.services.data_api_client')
     def test_shows_services_list(self, data_api_client):
         with self.app.test_client():
             self.login()
 
+            data_api_client.get_framework.side_effect = self._mock_get_framework_side_effect
             data_api_client.find_services.return_value = {
                 'services': [{
                     'serviceName': 'Service name 123',
                     'status': 'published',
                     'id': '123',
                     'lotSlug': 'saas',
                     'lotName': 'Software as a Service',
-                    'frameworkName': 'G-Cloud 1',
-                    'frameworkSlug': 'g-cloud-1'
+                    'frameworkName': 'G-Cloud 909',
+                    'frameworkSlug': 'g-cloud-909'
                 }]
             }
 
-            res = self.client.get('/suppliers/services')
+            res = self.client.get("/suppliers/frameworks/g-cloud-909/services")
             assert res.status_code == 200
+
+            document = html.fromstring(res.get_data(as_text=True))
+            assert document.xpath(
+                "//h1[normalize-space(string())=$t]",
+                t=u"Your G-Cloud 909 services",
+            )
+            assert document.xpath(
+                "//td[contains(@class, 'summary-item-field')][normalize-space(string())=$t]",
+                t=u"Software as a Service",
+            )
+            assert document.xpath(
+                "//td[contains(@class, 'summary-item-field')][normalize-space(string())=$t]",
+                t=u"Service name 123",
+            )
+
             data_api_client.find_services.assert_called_once_with(
-                supplier_id=1234)
-            assert "Service name 123" in res.get_data(as_text=True)
-            assert "Software as a Service" in res.get_data(as_text=True)
-            assert "G-Cloud 1" in res.get_data(as_text=True)
+                supplier_id=1234,
+                framework="g-cloud-909",
+            )
 
     @mock.patch('app.data_api_client')
     def test_should_not_be_able_to_see_page_if_made_inactive(self, services_data_api_client):
```

Comment:
Nitpick. If the test name said explitiely that `user` made in inactive then it's really easy to know when looking at the test name it's the user rather than a service being made inactive